aten/src/ATen/native/cuda/LossCTC.cu

```diff
@@ -302,18 +336,30 @@ ctc_loss_backward_log_beta_gpu_kernel(scalar_t* __restrict__ log_beta_data,
     int64_t current_target_prime;
     bool have_three;
     if (s < 2*target_length+1) {
-      current_target_prime = get_target_prime(targets_data, tg_batch_offset, tg_target_stride, s, BLANK);
-      have_three = ((s < 2*target_length-1) &&
-                    (get_target_prime(targets_data, tg_batch_offset, tg_target_stride, s+2, BLANK) !=
-                     current_target_prime));
+      current_target_prime = get_target_prime(
+          targets_data,
+          tg_batch_offset,
+          tg_target_stride,
+          s,
+          target_length,
+          BLANK);
+      have_three =
+          ((s < 2 * target_length - 1) &&
```

Comment:
Always false when `target_length == 0`

Reply:
I'll move this the target_length == 0 into the comments.